### PR TITLE
Alias BaseOperator.__mul__ to BaseOperator._multiply in quantum_info

### PR DESCRIPTION
**Pull request by daiki0623**
### Summary
Introduced a left side scalar multiplication to Operator class in quantum_info (issue: #8006 )

### Details and comments
Add `__mul__` method to Operator class to enable scalar multiplication not only from right side but left side.
It can be used by using * decorator,

For example
```
from qiskit.quantum_info import Pauli
Pauli("X") * -1j
```

### CODE REVIEW

File: qiskit/quantum_info/operators/mixins/multiply.py [hunk 1]
```diff
@@ -35,6 +35,9 @@ class MultiplyMixin(ABC):
     def __rmul__(self, other):
         return self._multiply(other)
 
+    def __mul__(self, other):
+        return self.__rmul__(other)
```
Comment (ikkoham) [inline qiskit/quantum_info/operators/mixins/multiply.py:39]: Using `__rmul__` looks a bit non-trivial, so how about using `_multiply` directly?

File: test/python/quantum_info/operators/symplectic/test_sparse_pauli_op.py [hunk 2]
```diff
@@ -13,6 +13,7 @@
 """Tests for SparsePauliOp class."""
 
 import itertools as it
+from tokenize import Number
```
Comment (ikkoham) [inline test/python/quantum_info/operators/symplectic/test_sparse_pauli_op.py:16]: Why did you add the new import?

**Review by ikkoham — CHANGES_REQUESTED** [review]
Thank you very much. I think it will be useful. I have a minor comment

File: qiskit/quantum_info/operators/mixins/multiply.py (hunk 1 above)
Comment (daiki0623) [reply qiskit/quantum_info/operators/mixins/multiply.py]: Thank you. Changed `__rmul__` to `_multiply`.

File: test/python/quantum_info/operators/symplectic/test_sparse_pauli_op.py (hunk 2 above)
Comment (daiki0623) [reply test/python/quantum_info/operators/symplectic/test_sparse_pauli_op.py]: Forgot to delete when developing. Thank you.

**Review by jakelishman — APPROVED** [review]
Looks good to me, thanks for this!  I left a small amount of historical context in https://github.com/Qiskit/qiskit-terra/issues/8006#issuecomment-1115958231 as a note to us in the future, but from my side this looks good to merge.  Ikko will be able to tag this "automerge" and it'll get merged.

**Review by ikkoham — APPROVED** [review]
LGTM! Thank you.
